src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -82,23 +172,56 @@ public List<GradeDTO> getAssignmentGrades(@PathVariable("assignmentId") int assi
         //   if the grade does not exist, create a grade entity and set the score to NULL
         //   and then save the new entity
 
-        return null;
+        Assignment a = assignmentRepository.findById(assignmentId).orElse(null);
+        if (a != null) {
+            int assignmentSectionId = a.getSection().getSectionNo();
+            List<Enrollment> enrollments = enrollmentRepository.findEnrollmentsBySectionNoOrderByStudentName(assignmentSectionId);
+
+            List<GradeDTO> grade_list = new ArrayList<>();
+
+            for (Enrollment e : enrollments) {
+                Grade grade = gradeRepository.findByEnrollmentIdAndAssignmentId(e.getEnrollmentId(), assignmentId);
+                if (grade != null) {
+                    grade_list.add(new GradeDTO(grade.getGradeId(), e.getUser().getName(), e.getUser().getEmail(), a.getTitle(),
+                            a.getSection().getCourse().getCourseId(), assignmentSectionId, grade.getScore()));
+                } else {
+                    Grade g = new Grade();
+                    g.setScore(null);
+                    g.setAssignment(assignmentRepository.findById(assignmentId).orElse(null));
+                    g.setEnrollment(e);
+                    grade_list.add(new GradeDTO(g.getGradeId(), e.getUser().getName(), e.getUser().getEmail(), a.getTitle(),
+                            a.getSection().getCourse().getCourseId(), assignmentSectionId, g.getScore()));
+                }
+            }
+            return grade_list;
+        }
+        else {
```

Comment:
Consider placing 
```
if(a == null) {
  throw new ResponseStatusException( HttpStatus.NOT_FOUND, "assignment not found ");
}
```
at the top to reduce how much code is inside an if bracket for readability. With this change `Else` is not needed as the code after a `ResponseStatusException` is not executed.